sys/fido2/doc.txt

```diff
@@ -0,0 +1,209 @@
+ *                     +-----------+  +-----------+   +-----------+   +-----|-----+
+ *                           |             |               |                |
+ *                           |             |               |                |
+ *                           |         +---------------------------+        |
+ *                           |         |                           |        |
+ *                           |---------|           ctap            |--------|
+ *                                     |                           |
+ *                                     +---------------------------+
+ *                                                   |
+ *                                     +-------------|-------------+
+ *                                     |                           |
+ *                                     |       ctap_transport      |
+ *                                     |                           |
+ *                                     +---------------------------+
+ *                                           |
+ *                                           |
+ *                                     +-----------+
+ *                                     |           |
+ *                                     | ctap_hid  |
+ *                                     |           |
+ *                                     +-----------+
+ *
+ *
+ *
+ * **ctap_hid**
+ *
+ * USB Human Interface Device (USB HID) transport binding for CTAP (CTAPHID).
+ *
+ * Initializes the USBUS HID interface.
+ *
+ * Communicates with the USBUS USB HID interface through the USBUS HID IO interface.
+ *
+ * **ctap_transport**
+ *
+ * Initializes CTAP layer.
+ *
+ * Initializes CTAP event queue.
+ *
+ * Manages CTAP transport bindings (currently only CTAPHID).
+ *
+ * **ctap**
+ *
+ * Contains the main CTAP logic.
+ *
+ * Makes use of helpers for flash access, cryptographic operations and CBOR operations.
+ *
+ * **ctap_cbor**
+ *
+ * Helper containing functionality to parse and encode CBOR messages. Uses the [tinyCBOR](https://doc.riot-os.org/group__pkg__tinycbor.html) pkg.
+ *
+ * **ctap_crypto**
+ *
+ * Helper containing functionality for cryptographic operations.
+ *
+ * Abstraction for native cryptographic operations (SHA256, HMAC-SHA-256, AES CCM).
+ *
+ * @warning This abstraction ignores the pointer to a static buffer returned by some functions (e.g. @ref sha256). Therefore a valid pointer to a buffer for the digest always needs to be provided when necessary.
+ *
+ * Abstraction for Elliptic curve cryptography (ECC) operations. Uses the [micro-ecc](https://api.riot-os.org/group__pkg__micro__ecc.html) pkg.
+ *
+ * Parsing of cryptographic signatures into ASN.1 DER format. Uses the [tiny-asn1](http://doc.riot-os.org/group__pkg__tiny-asn1.html) pkg.
+ *
+ * **ctap_mem**
+ *
+ * Abstraction for native flash operations based on [Flash page driver](https://doc.riot-os.org/group__drivers__periph__flashpage.html).
```

Comment:
What means "native"? Maybe mention that you write the keys "raw" (without any filesystem or format)

Reply:
Native because it uses the RIOT flashpage API.

Reply:
Then say so :)!